### ISSUE TITLE
OpenSSL.SSL.Context missing from Intersphinx mapping in 21.0.0

### ISSUE THREAD

**Issue by pquentin** [pyca/pyopenssl#1046]
# Steps to reproduce

Inspect published Sphinx mapping:

```bash
$ curl https://www.pyopenssl.org/en/21.0.0/objects.inv > objects.inv && python -m sphinx.ext.intersphinx objects.inv | grep Context
        OpenSSL.SSL.ContextType                  api/ssl.html#OpenSSL.SSL.ContextType
        openssl-context                          Context objects                         : api/ssl.html#openssl-context
        openssl-x509storecontext                 X509StoreContext objects                : api/crypto.html#openssl-x509storecontext
        openssl-x509storecontexterror            X509StoreContextError objects           : api/crypto.html#openssl-x509storecontexterror
```

# Expected output

Just like 20.0.1:

```bash
$ curl https://www.pyopenssl.org/en/20.0.1/objects.inv > objects.inv && python -m sphinx.ext.intersphinx objects.inv | grep Context   
 https://www.pyopenssl.org/en/stable
 https://www.pyopenssl.org/en/stable
	OpenSSL.SSL.Context                      api/ssl.html#OpenSSL.SSL.Context
	OpenSSL.crypto.X509StoreContext          api/crypto.html#OpenSSL.crypto.X509StoreContext
	OpenSSL.crypto.X509StoreContextError     api/crypto.html#OpenSSL.crypto.X509StoreContextError
	OpenSSL.SSL.ContextType                  api/ssl.html#OpenSSL.SSL.ContextType
	OpenSSL.SSL.Context.add_client_ca        api/ssl.html#OpenSSL.SSL.Context.add_client_ca
    ...
	OpenSSL.SSL.Context.use_privatekey_file  api/ssl.html#OpenSSL.SSL.Context.use_privatekey_file
	OpenSSL.crypto.X509StoreContext.get_verified_chain api/crypto.html#OpenSSL.crypto.X509StoreContext.get_verified_chain
	OpenSSL.crypto.X509StoreContext.set_store api/crypto.html#OpenSSL.crypto.X509StoreContext.set_store
	OpenSSL.crypto.X509StoreContext.verify_certificate api/crypto.html#OpenSSL.crypto.X509StoreContext.verify_certificate
	openssl-context                          Context objects                         : api/ssl.html#openssl-context
	openssl-x509storecontext                 X509StoreContext objects                : api/crypto.html#openssl-x509storecontext
	openssl-x509storecontexterror            X509StoreContextError objects           : api/crypto.html#openssl-x509storecontexterror
```

**Comment by njsmith** [comment]
Status note before going to bed:

- We did just fix a sphinx warning about having two links for `Context`, and needing to add `:noindex:` to one of them. So that might be related?
- But we do have an `.. autoclass:: OpenSSL.SSL.Context` that doesn't show up in the RTD index, even though it ought to (we only added `:noindex:` to one of the instances, not the other)
- When I build locally, I *do* get an index entry for `Context`, as well as many many other things that are missing from the RTD index
- One difference between my local build and the RTD build is that RTD is for some reason insisting on using an ancient version of sphinx (it does `pip install sphinx<2`; latest version is 4.2.0). That might be the problem?
- We have not figured out why RTD is doing this, or how to get it to build PRs so we can more easily experiment with changing it
